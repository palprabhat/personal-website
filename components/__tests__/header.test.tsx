import { render, screen } from "@testing-library/react";
import Header from "../header";
import { headerTestId, logoId, myEmail } from "../../constants/testIds.const";

describe("Header Component", () => {
  it("renders correct content", () => {
    render(<Header />);

    const header = screen.queryByTestId(headerTestId);
    const image = screen.queryByTestId(logoId);
    const email = screen.queryByTestId(myEmail);

    expect(header).toBeInTheDocument();
    expect(header).toContainElement(image);
    expect(header).toContainElement(email);
  });
});
