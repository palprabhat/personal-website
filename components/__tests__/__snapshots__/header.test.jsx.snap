// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correct content 1`] = `
<body>
  <div>
    <div
      class="mx-12 mt-3 hidden md:flex justify-between"
      data-testid="header-test-id"
    >
      <div
        style="max-width: 100%; width: 72px;"
      >
        <div
          style="position: relative; padding-bottom: 100%;"
        >
          
          <img
            alt="logo"
            class="__lazy"
            data-src="/_next/image?url=%2Fimages%2Flogo.svg&w=320&q=75"
            data-srcset="/_next/image?url=%2Fimages%2Flogo.svg&w=320&q=75 320w"
            data-testid="logo-id"
            style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
          />
        </div>
      </div>
      <a
        class="flex justify-center items-center px-2 group"
        data-testid="my-email"
        href="mailto:prabhatpal.14@gmail.com"
      >
        <span
          class="brand-icon-email mr-1 group-hover:text-primary-600 group-focus:text-primary-600"
        />
        prabhatpal.14@gmail.com
      </a>
    </div>
  </div>
</body>
`;
